modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/GetEntryRenderDataMVCResourceCommand.java

```diff
@@ -1129,9 +1129,19 @@ private <T extends BaseModel<T>> JSONObject _getWorkflowDataJSONObject(
 		long currentCTCollectionId =
 			CTCollectionThreadLocal.getCTCollectionId();
 
+		CTCollection ctCollection = _ctCollectionLocalService.getCTCollection(
+			ctEntry.getCtCollectionId());
+
+		long safeCloseableCTCollectionId = ctEntry.getCtCollectionId();
+
+		if (ctCollection.getStatus() != WorkflowConstants.STATUS_DRAFT) {
+			safeCloseableCTCollectionId =
+				CTCollectionThreadLocal.getCTCollectionId();
+		}
+
 		try (SafeCloseable safeCloseable =
 				CTCollectionThreadLocal.setCTCollectionIdWithSafeCloseable(
-					ctEntry.getCtCollectionId())) {
+					safeCloseableCTCollectionId)) {
```

Comment:
Can you please clarify the expected behavior for this change @ctang3?

When a publication is published and we are viewing the workflow information in the history, do we want to see the status of the entity in the current publication or in production only?

Reply:
I think seeing the entity status in the current publication makes the most intuitive sense for users, since they would be accustomed to the idea that things have different states in different publications; so if it were to show the status in production only, that behavior would not align with the general convention. I think if we want to show the status in production only, we should add some sort of notice to the user about that. 

If I'm remembering right, I originally made this change (prior to your LPD-22771 getting merged) because there was an unexpected error when trying to view the already-published CTEntry's Workflow Tab - `WorkflowInstanceLink workflowInstanceLink = _getWorkflowInstanceLink(ctEntry, model);` was returning null due to the CTCollection scoping to the ctEntry's ctCollectionId.

I also just tested whether the Workflow Tab shows without error if I view the Workflow Tab from a newly created Publication.

Reply:
Ok, that makes sense to me. Thanks @ctang3!

Reply:
@ctang3 The workflow information should be consistent with the info that is shown in the Data tab. The status value is retrieved from the `model` object, but the workflow info is retrieved from the WorkflowInstance/WorkflowTask objects. That creates a discrepancy in the info displayed where the status is read from production and workflow data (ie. assignee) is read from the current publication when the entity is added and published in a publication.

To test the scenario:
1. Enable workflow for blogs
2. Add publication p1
3. Add blog1 in p1
4. Assign to Test Test and approve the blog
5. Publish p1
6. View p1 in history. Assert blog1 has workflow status APPROVED and assignee Test Test
7. Add publication p2
8. Modify blog1 in p2 and submit it for workflow. The workflow status for blog1 should now be PENDING and assignee is NOBODY.
9. View p1 in history. Assert blog1 workflow status is APPROVED and assignee is NOBODY (note that the status is APPROVED because it's read from production and the assignee here is NOBODY because it's getting from the workflow task in p2)

We should use the ctCollectionId from the `model` to have consistent data.

Reply:
@pei-jung The ctCollectionId of the `model` was the production id. I initially thought to change the `T model` parameter to `CTModel<T> model` for `_getWorkflowDataJSONObject()` so that I can get the CTCollectionId from `model`, but that would require a lot of changes. 

I decided to go with fetching the corresponding CTModel for the current ctCollectionId using CTDisplayRendererRegistry and using that as the model to get the status field from. Is this an acceptable solution? If so, I'll squash this change and the one to move the check in your other comment down into 1 commit for forwarding. Thanks!

Reply:
We can get the ctCollectionId from the model attributes `model.getModelAttributes()`

Reply:
I checked and the ctCollectionId from that is still 0, since the `model` arg passed into `_getWorkflowDataJSONObject()` is the production model. So the status if using that would be whatever is on Production, not the current Publication. 

```
Map<String, Object> modelAttributes = model.getModelAttributes();

modelAttributes.get("ctCollectionId"); // returns 0 for production
```

To confirm - when viewing P1 in another publication P2, do we want the status and asignee to be the values from P2 when viewing the workflow tab? My understanding is that's the behavior we want, which I think https://github.com/liferay-publications/liferay-portal/commit/33905b66473dd8024bc17423b988ad8ea11b9367 should accomplish?